### ISSUE TITLE
Support for header-only libraries in the blt_add_library() macro

### ISSUE THREAD

**Issue by gzagaris** [LLNL/blt#5]
The following currently does not work:
```
blt_add_library(
    NAME                                      <mylib>
    HEADERS                                ${mylib_headers}
    HEADERS_OUTPUT_SUBDIR <mylib_dir>
    DEPENDS_ON                          <mylib_depends_on>
    )
```

`blt_add_library` requires a non-empty `SOURCES` parameter

**Comment by rhornung67** [comment]
@gzagaris and I discussed this earlier today. What do folks think about adding seprate options for TPL dependencies vs. project dependencies; e.g., DEPENDS_ON_TPL and DEPENDS_ON_WITHIN_PROJECT (or whatever makes sense). It seems that we want to discrimiate between intoerproject dependencires (e.g., between axom components and between TPLs).

**Comment by gzagaris** [comment]
@rhornung67 I would vote for `TPL_DEPENDS` to list TPL dependencies and keep `DEPENDS_ON` for other cmake identifiable targets within the same project/superproject (e.g., axom components). This would also simplify some of the logic internally where we have to do stuff like:

`if (TARGET foo)`
 `     # foo is a component `
`endif()`

**Comment by davidbeckingsale** [comment]
Don't we have enough information to determine whether something is a TPL or not without two lists?

**Comment by gzagaris** [comment]
@davidbeckingsale We do, but that bloats the cmake code with if/else logic that can be avoided if we explicitly list the TPLs. Also, things like `list(FIND arg_DEPENDS_ON "cuda" check_for_cuda)` will be more efficient if the list is smaller, e.g., `list(FIND arg_TPL_DEPENDS...)

Also, from a maintenance perspective, listing explicitly the TPLs vs. other targets within the same project/sub-project seems like a good thing.

**Comment by cyrush** [comment]
we talked about this at a project weekly meeting. 

We don't need to have separate TPL_DEPENDS -- full transitivity support will resolve this issue eventually. 


Eventually, a target should only have to list deps for the things it is directly using.

If that list includes another target that has extended deps (via headers or libs), regardless if that dep is a TPL or one of our targets (it shouldn't matter), CMake will handle it for us if we set things up correctly. 


@davidbeckingsale  I think the existing solution you had would well. Can you resurrect it?

**Comment by davidbeckingsale** [comment]
Yup, I'm on it.

**Comment by davidbeckingsale** [comment]
Closed by #16
